### PR TITLE
454 -  Change Evidence of connectivity speeds upload

### PR DESCRIPTION
**Pull request by marcellmueller**
https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/454

This PR also reverts the other supporting documents schema back to an object and not a function as that conditional required isn't needed anymore.

### CODE REVIEW

File: app/formSchema/schema.ts [hunk 1]
```diff
@@ -22,10 +22,7 @@ import {
   templateUploads,
 } from './pages';
 
-const useSchema = (formData) => {
-  const isEvidenceOfConnectivityRequired =
-    formData?.templateUploads?.supportingConnectivityEvidence;
-
+const useSchema = () => {
```
Comment (matthieu-foucault) [inline app/formSchema/schema.ts:25]: can schema be an exported constant? That will avoid recreating a new schema object all the time and triggering unnecessary re-renders

File: app/formSchema/uiSchema.ts [hunk 2]
```diff
@@ -654,10 +653,9 @@ const uiSchema = {
...
     },
   },
-  evidenceOfConnectivitySpeeds: {
+  otherSupportingMaterials: {
     'ui:widget': 'FileWidget',
-    'ui:description':
-      'Evidence of connectivity speeds (e.g, screen captures of Speed Test results)',
+    'ui:description': 'Other supporting materials (if applicable)',
     'ui:options': {
       maxLength: MAX_LONG_INPUT_LENGTH,
```
Comment (matthieu-foucault) [inline app/formSchema/uiSchema.ts:660]: do we need an option to hide the "optional" keyword?

File: app/formSchema/pages/supportingDocuments.ts [hunk 3]
```diff
@@ -1,55 +1,49 @@
...
+      },
+      communityRuralDevelopmentBenefits: {
+        title:
+          'Please refer to Annex 3 of the application guide for the community letters of support requirements. If you do not currently have these, you will have a 30-day grace period after the intake closes to submit them.',
+        type: 'string',
+      },
+      otherSupportingMaterials: {
+        title:
+          'Please upload any other files such as evidence of connectivity speeds (such as screen captures of speed test results), a written commitment to facilitating access to Passive Infrastructure, evidence of imminent access to Third Party infrastructure, evidence of other funding sources, network information and/or coverage for ISED, or other documents to support this application.',
+        type: 'string',
```
Comment (matthieu-foucault) [inline app/formSchema/pages/supportingDocuments.ts:43]: should our schemas ask for an array of strings when we allow multiple uploads? How are we storing the uploaded files ids in the json data?

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: app/formSchema/uiSchema.ts (hunk 2 above)
Comment (wenzowski) [reply app/formSchema/uiSchema.ts:660]: any chance we could implement as a string label to render for optional? This way we'd be enforcing that all optional fields have some label even if it isn't `(optional)`

Comment (marcellmueller) [reply app/formSchema/uiSchema.ts:660]: > any chance we could implement as a string label to render for optional? This way we'd be enforcing that all optional fields have some label even if it isn't `(optional)`

That's well out of scope of this ticket as well as there are some fields iirc that are not required but they wanted optional hidden.

Comment (marcellmueller) [reply app/formSchema/uiSchema.ts:660]: > do we need an option to hide the "optional" keyword?

good catch, I did that and renamed `ignoreOptional` to `hideOptional` to make it a bit more clear

File: app/formSchema/pages/supportingDocuments.ts (hunk 3 above)
Comment (matthieu-foucault) [reply app/formSchema/pages/supportingDocuments.ts:43]: This is out of scope but I'd love to discuss this at some point

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**